java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -757,11 +757,11 @@ private void executeTypeCast(TypeCastTree typeCast) {
   }
 
   private void executeAssignement(AssignmentExpressionTree tree) {
-    ExpressionTree variable = tree.variable();
     ProgramState.Pop unstack;
     SymbolicValue value;
 
-    if (ExpressionUtils.isSimpleAssignment(tree)) {
+    boolean isSimpleAssignment = ExpressionUtils.isSimpleAssignment(tree);
+    if (isSimpleAssignment) {
       unstack = programState.unstackValue(1);
```

Comment:
You probably want to unstack also the SV associated to `this` or `super` in case of assignment using them, to only keep the sv associated to the field. I'm guessing it's what is currently having impact on compounds assignments for the division by zero.